src/common/ServerCompatBanner.js

```diff
@@ -58,50 +29,50 @@ export default function ServerCompatBanner(props: Props): Node {
   const isAdmin = useSelector(getIsAdmin);
   const settings = useSelector(getSettings);
 
+  let visible = false;
+  let text = '';
   if (!zulipVersion || zulipVersion.isAtLeast(minSupportedVersion)) {
-    return null;
+    // don't show
   } else if (hasDismissedServerCompatNotice) {
-    return null;
+    // don't show
+  } else {
+    visible = true;
+    text = isAdmin
+      ? {
+          text:
+            '{realm} is running Zulip Server {serverVersion}, which is unsupported. Please upgrade your server as soon as possible.',
+          values: { realm: realm.toString(), serverVersion: zulipVersion.raw() },
+        }
+      : {
+          text:
+            '{realm} is running Zulip Server {serverVersion}, which is unsupported. Please contact your administrator about upgrading.',
+          values: { realm: realm.toString(), serverVersion: zulipVersion.raw() },
+        };
   }
 
   return (
-    <SafeAreaView mode="padding" edges={['right', 'left']} style={styles.wrapper}>
-      <View style={styles.textRow}>
-        <Label
-          style={styles.text}
-          text={
-            isAdmin
-              ? {
-                  text:
-                    '{realm} is running Zulip Server {serverVersion}, which is unsupported. Please upgrade your server as soon as possible.',
-                  values: { realm: realm.toString(), serverVersion: zulipVersion.raw() },
-                }
-              : {
-                  text:
-                    '{realm} is running Zulip Server {serverVersion}, which is unsupported. Please contact your administrator about upgrading.',
-                  values: { realm: realm.toString(), serverVersion: zulipVersion.raw() },
-                }
-          }
-        />
-      </View>
-      <View style={styles.buttonsRow}>
-        <ZulipTextButton
-          label="Dismiss"
-          onPress={() => {
+    <ZulipBanner
+      visible={visible}
+      text={text}
+      buttons={[
+        {
+          id: 'dismiss',
+          label: 'Dismiss',
+          onPress: () => {
             dispatch(dismissCompatNotice());
-          }}
-        />
-        <ZulipTextButton
-          leftMargin
```

Comment:
This `leftMargin` gets dropped, in the `[nfc]` commit that factors out `ZulipBanner`.

Did we need it? It looked like it was there to provide spacing between the two buttons.

Probably using flexbox is better, and I see that's the direction the later commits aim at. If it's a pain to reproduce this behavior in the initial factoring-out, then I don't mind having the layout a bit wonky as an intermediate state; just mention in the commit message.

Reply:
OK, having now read the remaining commits I see they don't appear to affect this between-button spacing. And looking at the screenshots above, I see there was a change. So I think a fix to keep this spacing is needed.

Looking at the available flexbox properties, I think those won't actually give us what we want: 8px between items. Which maybe makes sense because after all that isn't really about *flex*ing.

So probably the simplest thing is to keep using `leftMargin`. You can use the second parameter to the `Array#map` callback (i.e., the index) to add it back without much change to the new code.